server/test/write.go

```diff
@@ -1398,40 +1407,10 @@ var writeCommandTests = []writeCommandTest{
 				tuple.NewTupleKey("document:budget", "parent", "folder:budgets#admin"),
 			}},
 		},
-		err: serverErrors.InvalidTuple("Userset 'folder:budgets#admin' is not allowed to have relation 'parent' with 'document:budget'",
+		err: serverErrors.InvalidTuple("unexpected userset relationship 'folder:budgets#admin' with tupleset relation 'document#parent'",
 			tuple.NewTupleKey("document:budget", "parent", "folder:budgets#admin"),
 		),
 	},
-	{
```

Comment:
This test is an invalid test since the `folder#parent` relation is a tupleset relation and is defined under a Union rewrite.

Reply:
I don't understand - the tuple being written in this test is referencing the `parent` relation of the `document` type, which is defined as `self`

Reply:
🤔 I guess I'm confused what this is testing then. At face value it looks wrong and is confusing because `docuemnt#parent` is often referred to in a tupleset relation and so writing a userset value to it seems incorrect.

Now that I look at it after having a break though, it appears as though this is just testing that if a `parent` tupleset relation exists somewhere else in the model on some other type then the write is OK. Was that the intent @miparnisari ?

Reply:
Yes, I think so. I remember while developing this code that, when retrieving all tupleset relations, I had forgotten to check that the type of the relation matched the type of the tuple.

Reply:
Alright, cool. I'll add it back.